ideascube/tests/test_import_users.py

```diff
@@ -0,0 +1,180 @@
+    response = form.submit()
+    response.follow()
+    assert User.objects.count() == 2
+    user = User.objects.order_by('pk').last()
+    assert user.serial == '12345'
+    assert user.full_name == 'جبران خليل جبران'
+
+
+def test_should_import_boolean(staffapp, monkeypatch):
+    # True/False is how we do export the data, so let's accept this in input
+    # too.
+    data = (b'serial,full_name,is_sent_to_school\n'
+            b'12345,Gabriel Garcia Marquez,True\n'
+            b'23456,Miguel de Cervantes,False')
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    assert User.objects.count() == 3
+    user = User.objects.order_by('pk')[1]
+    assert user.serial == '12345'
+    assert user.is_sent_to_school is True
+    user = User.objects.order_by('pk')[2]
+    assert user.serial == '23456'
+    assert user.is_sent_to_school is False
+
+
+def test_should_import_date(staffapp, monkeypatch):
+    data = (b'serial,full_name,camp_entry_date\n'
+            b'12345,Gabriel Garcia Marquez,2016-10-27')
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    assert User.objects.count() == 2
+    user = User.objects.order_by('pk').last()
+    assert user.serial == '12345'
+    assert user.camp_entry_date.year == 2016
+    assert user.camp_entry_date.month == 10
+    assert user.camp_entry_date.day == 27
+
+
+def test_should_update_existing_users(staffapp, monkeypatch):
+    data = (b'serial,full_name\n'
+            b'12345,Gabriel Garcia Marquez')
+    UserFactory(full_name='Gabriel Garcia Marques', serial='12345')
+    assert User.objects.count() == 2
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    assert User.objects.count() == 2
+    user = User.objects.order_by('pk').last()
+    assert user.serial == '12345'
+    assert user.full_name == 'Gabriel Garcia Marquez'  # s has been fixed in z.
+
+
+def test_should_not_replace_is_staff(staffapp, monkeypatch):
+    data = (b'serial,full_name,is_staff\n'
+            b'12345,Gabriel Garcia Marquez,True\n'
+            b'23456,Miguel de Cervantes,False')
+    UserFactory(serial='12345', is_staff=False)
+    UserFactory(serial='23456', is_staff=True)
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    user = User.objects.order_by('pk')[1]
+    assert user.serial == '12345'
+    assert user.full_name == 'Gabriel Garcia Marquez'
+    assert user.is_staff is False
+    user = User.objects.order_by('pk')[2]
+    assert user.is_staff is True
+    assert user.full_name == 'Miguel de Cervantes'
```

Comment:
> Change the password for the __system__ user, making it possible for someone to login with its credentials?

We are not taking password into account, so I think we are safe on this side (just added some unittests to cover the password scenario). We may want at some point take the password into account to be able to import even the staff users, but that's out of current scope.

> Yes, it will fail. As the system user is filtered, it tries to create a new system user and fails with a duplicate primary key error.

Fixed and added a test to cover the scenario.